### ISSUE TITLE
Can't load huggingface model with ktrain 0.33

### ISSUE THREAD

**Issue by johngcolson** [amaiya/ktrain#478]
Hi, ktrain new version (0.33) no longer makes it possible to load huggingface model
dbmdz/electra-large-discriminator-finetuned-conll03-english

(see error code below)
Importing version 0.32 of ktrain instead of 0.33 does load the model.
------------------------------------------------------------------------------------------------------------------------------------------
OSError                                   Traceback (most recent call last)
[<ipython-input-7-787c44aa4051>](https://localhost:8080/#) in <module>
      3 from ktrain import text
      4 MODEL_NAME = "dbmdz/electra-large-discriminator-finetuned-conll03-english"
----> 5 t = text.Transformer(MODEL_NAME, maxlen=45, class_names=categories)
      6 trn = t.preprocess_train(x_train, y_train)

3 frames
[/usr/local/lib/python3.8/dist-packages/transformers/modeling_tf_utils.py](https://localhost:8080/#) in from_pretrained(cls, pretrained_model_name_or_path, *model_args, **kwargs)
   2667                         }
   2668                         if has_file(pretrained_model_name_or_path, WEIGHTS_NAME, **has_file_kwargs):
-> 2669                             raise EnvironmentError(
   2670                                 f"{pretrained_model_name_or_path} does not appear to have a file named"
   2671                                 f" {TF2_WEIGHTS_NAME} but there is a file for PyTorch weights. Use `from_pt=True` to"

OSError: dbmdz/electra-large-discriminator-finetuned-conll03-english does not appear to have a file named tf_model.h5 but there is a file for PyTorch weights. Use `from_pt=True` to load this model from those weights.

**Comment by johngcolson** [comment]
Thank you, here are the details:

Version of tensorflow:  2.9.2
Python 3.8.10
transformers 4.26.0

**Comment by amaiya** [comment]
Hi @johngcolson 

This doesn't appear to be an issue with **ktrain**, as the problem occurs when using **transformers** directly (without **ktrain**):

```
In [2]: from transformers import TFAutoModelForSequenceClassification

In [3]: model = TFAutoModelForSequenceClassification.from_pretrained('dbmdz/electra-large-discriminator-finetuned-conll03-english')
Downloading: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 828/828 [00:00<00:00, 564kB/s]
---------------------------------------------------------------------------
OSError                                   Traceback (most recent call last)
Input In [3], in <cell line: 1>()
----> 1 model = TFAutoModelForSequenceClassification.from_pretrained('dbmdz/electra-large-discriminator-finetuned-conll03-english')

File ~/mambaforge/lib/python3.9/site-packages/transformers/models/auto/auto_factory.py:463, in _BaseAutoModelClass.from_pretrained(cls, pretrained_model_name_or_path, *model_args, **kwargs)
    461 elif type(config) in cls._model_mapping.keys():
    462     model_class = _get_model_class(config, cls._model_mapping)
--> 463     return model_class.from_pretrained(
    464         pretrained_model_name_or_path, *model_args, config=config, **hub_kwargs, **kwargs
    465     )
    466 raise ValueError(
    467     f"Unrecognized configuration class {config.__class__} for this kind of AutoModel: {cls.__name__}.\n"
    468     f"Model type should be one of {', '.join(c.__name__ for c in cls._model_mapping.keys())}."
    469 )

File ~/mambaforge/lib/python3.9/site-packages/transformers/modeling_tf_utils.py:2642, in TFPreTrainedModel.from_pretrained(cls, pretrained_model_name_or_path, *model_args, **kwargs)
   2636 has_file_kwargs = {
   2637     "revision": revision,
   2638     "proxies": proxies,
   2639     "use_auth_token": use_auth_token,
   2640 }
   2641 if has_file(pretrained_model_name_or_path, WEIGHTS_NAME, **has_file_kwargs):
-> 2642     raise EnvironmentError(
   2643         f"{pretrained_model_name_or_path} does not appear to have a file named"
   2644         f" {TF2_WEIGHTS_NAME} but there is a file for PyTorch weights. Use `from_pt=True` to"
   2645         " load this model from those weights."
   2646     )
   2647 else:
   2648     raise EnvironmentError(
   2649         f"{pretrained_model_name_or_path} does not appear to have a file named"
   2650         f" {TF2_WEIGHTS_NAME} or {WEIGHTS_NAME}."
   2651     )

OSError: dbmdz/electra-large-discriminator-finetuned-conll03-english does not appear to have a file named tf_model.h5 but there is a file for PyTorch weights. Use `from_pt=True` to load this model from those weights.
```

I downgraded **transformers** to v4.17.0 (which was used by the older version of **ktrain**) and it also doesn't load.  

Are you sure this model worked with `ktrain<0.33.0`?

**Comment by johngcolson** [comment]
Hi, yes it works with ktrain 0.32. 

!pip install -q ktrain==0.32.0
...
With this version of ktrain, there is just a warning 

"/usr/local/lib/python3.8/dist-packages/ktrain/text/preprocessor.py:1170: _UserWarning: Could not load a Tensorflow version of model. (If this worked before, it might be an out-of-memory issue.) Attempting to download/load PyTorch version as TensorFlow model using from_pt=True. You will need PyTorch installed for this."

but the model loads, as can be seen below:

![temp](https://user-images.githubusercontent.com/124262161/216548829-8ae6f185-c176-4df1-bac3-ce28d76328c8.jpg)

**Comment by amaiya** [comment]
I see, so the older version works by loading the PyTorch version successfully.  Thanks for posting issue - I'll look into this further to see what's going on.

**Comment by amaiya** [comment]
This should be resolved in `ktrain==0.33.1`.  If you have further problems, please reply to this issue. Thanks again for opening the issue to let us know.

**Comment by johngcolson** [comment]
Thank you, all is indeed working fine.

Ktrain is a great wrapper.

Best,

John G. Colson

Le ven. 3 févr. 2023 à 20:28, Arun S. Maiya ***@***.***> a
écrit :
